src/components/oculus-touch-controls.js

```diff
@@ -320,20 +357,122 @@ module.exports.Component = registerComponent('oculus-touch-controls', {
     });
   },
 
+  applyOffset: function (model) {
+    model.position.copy(this.displayModel[this.data.hand].modelPivotOffset);
+    model.rotation.copy(this.displayModel[this.data.hand].modelPivotRotation);
+  },
+
+  multiMeshFix: function (button) {
+    if (!this.buttonMeshes[button]) return;
+    var self = this;
+    this.buttonMeshes[button].traverse(function (node) {
+      if (node.type !== 'Mesh') return;
+      let newMaterial = node.material.clone();
+      self.buttonMeshes[button].naturalColor = node.material.color;
+      node.material.dispose();
+      node.material = newMaterial;
+    });
+  },
+
+  onOculusTouchV3ModelLoaded: function (evt) {
+    var controllerObject3D = this.controllerObject3D = evt.detail.model;
+
+    if (!this.data.model) { return; }
```

Comment:
this one could be removed if we do https://github.com/aframevr/aframe/pull/5103/files#r966498190

Reply:
Yup, also noticed that, and agreed.